cupy/core/fusion.pyx

```diff
@@ -729,15 +730,30 @@ class _FusionHistory(object):
             The second element of return values is kwargs that will give into
             the elementwise kernel or reduction kernel.
         """
-        in_dtypes = [t for t, d in in_params_info]
-        in_ndims = [d for t, d in in_params_info]
-        self.ndim = max(in_ndims)
-        in_params = [self._fresh_premap_param(t) for t in in_dtypes]
-        in_pvars = [_FusionVarScalar(v, d, False)
-                    if d == -1
-                    else _FusionVarArray(v, d, False)
-                    for v, d in zip(in_params, in_ndims)]
-        return_value = func(*in_pvars)
+
+        def gen_abstracted_args(a):
+            if isinstance(a, core.ndarray):
+                cuda_var = self._fresh_premap_param(a.dtype)
+                python_var = _FusionVarArray(cuda_var, a.ndim, False)
+            elif a is None:
+                cuda_var = None
+                python_var = None
+            else:
+                cuda_var = self._fresh_premap_param(numpy.dtype(type(a)))
+                python_var = _FusionVarScalar(cuda_var, -1, False)
+            return cuda_var, python_var
+
+        self.ndim = max([a.ndim for a in args if isinstance(a, core.ndarray)])
```

Comment:
Should we change the name of the input argument in the docs from `in_types` to `args` as well?